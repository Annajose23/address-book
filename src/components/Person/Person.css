.person {
    width: 500px;
    padding: 16px;
    text-align: center;
    border: 1px solid #eee;
    display: inline-block;
    margin: 10px;
    box-sizing: border-box;
    cursor: pointer;
    border-radius: 5px;
}

.person:hover,
.person:active {
    background-color: #242526;
}

