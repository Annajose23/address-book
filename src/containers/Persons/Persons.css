.address_book {
    display: flex;
    flex-flow: row wrap;
    justify-content: center;
    width: 80%;
    margin: auto;
}

div>h3{
    text-align: center;
    
}

@media(min-width:40rem){
    .address_book{
        width: 100%;
    }
}
