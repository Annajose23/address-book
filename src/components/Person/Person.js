import React from "react";
import './Person.css';

const person = (props) => {
  return (
    <article className="person" onClick={props.clicked}>
        {props.name.title}. {props.name.first} {props.name.last}
    </article>
  );
};

export default person;
