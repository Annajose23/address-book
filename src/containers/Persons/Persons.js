import React, { Component } from "react";
import Person from "../../components/Person/Person";
import { connect } from "react-redux";

import * as actionCreators from "../../store/actions/actions";
import Aux from "./../../hoc/Auxilary/Auxilary";
import Spinner from "../../components/UI/Spinner/Spinner";
import "./Persons.css";

export class Persons extends Component {
  
  componentDidMount() {
    this.props.onFetchPersons();
  }

  contactSelectHandler = (person) => {
    this.props.onContactSelection(person);
    this.props.history.replace("/contact-details");
  };

  render() {
    let persons = this.props.personsList.map((person, index) => (
      <Person
        key={index}
        name={person.name}
        clicked={() => this.contactSelectHandler(person)}
      />
    ));

    if (this.props.loading) {
      persons = <Spinner />;
    }

    if (this.props.hasError) {
      persons = <p>Something went wrong!!</p>;
    }

    return (
      <Aux>
        <div>
          <h3>Address Book</h3>
        </div>
        <div className="address_book">{persons}</div>
      </Aux>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    personsList: state.persons,
    loading: state.loading,
    hasError: state.hasError,
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    onFetchPersons: () => dispatch(actionCreators.fetchPersons()),
    onContactSelection: (person) =>
      dispatch(actionCreators.selectPersonSuccess(person)),
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(Persons);
