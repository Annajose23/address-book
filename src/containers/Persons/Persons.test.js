import React from 'react';
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import renderer from 'react-test-renderer'; 

import Aux from "../../hoc/Auxilary/Auxilary";
import {Persons} from './Persons';
import Person from '../../components/person/person';
import Spinner from '../../components/UI/Spinner/Spinner';

configure({ adapter: new Adapter() });

describe("<Persons />", () => {
    let wrapper;

    let props = {
        personsList: [{
            name: {
              title: "Ms",
              first: "Anna",
              last: "Jose",
            },
            cell: "0123456"}],
        loading: false,
        hasError: false,
      };
    
    beforeEach(() => {
      wrapper = shallow(<Persons {...props} onFetchPersons={()=>{}} onContactSelection={()=>{}}/>);
    });
  
    it("should render <Person />", () => {
      expect(wrapper.find(Person)).toHaveLength(1);
    });

    it("should contain <Aux />", () => {
      expect(wrapper.find(Aux)).toHaveLength(1);
    });

    it("should contain <Spinner />", () => {
      wrapper.setProps({loading:true});
      expect(wrapper.find(Spinner)).toHaveLength(1);
    });

    it("should throw error if fetch persons failed", () => {
      wrapper.setProps({hasError:true});
      expect(wrapper.contains(<p>Something went wrong!!</p>)).toEqual(true);
    });

    it('Should contain two div', () => {
      expect(wrapper.find('div').length).toEqual(2);
  });
  
    it("render correctly Persons component", () => {
      const PersonsComponent = renderer
        .create(<Persons {...props} onFetchPersons={()=>{}} onContactSelection={()=>{}}/>)
        .toJSON();
      expect(PersonsComponent).toMatchSnapshot();
    });
  });
  