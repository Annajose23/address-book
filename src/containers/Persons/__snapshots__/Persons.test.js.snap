// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Persons /> render correctly Persons component 1`] = `
Array [
  <div>
    <h3>
      Address Book
    </h3>
  </div>,
  <div
    className="address_book"
  >
    <article
      className="person"
      onClick={[Function]}
    >
      Ms
      . 
      Anna
       
      Jose
    </article>
  </div>,
]
`;
